tw-tasks-core/src/main/java/com/transferwise/tasks/utils/TwTasksUuidUtils.java

```diff
@@ -0,0 +1,31 @@
+package com.transferwise.tasks.utils;
+
+import com.transferwise.common.baseutils.UuidUtils;
+import java.math.BigInteger;
+import java.util.UUID;
+import lombok.experimental.UtilityClass;
+import org.apache.commons.lang3.NotImplementedException;
+import org.apache.commons.lang3.StringUtils;
+
+@UtilityClass
+public class TwTasksUuidUtils {
+
+  public static UUID toUuid(String st) {
```

Comment:
This is in purpose to support UUIDs that are not supported by `UUID.formString` or maybe `UUID.fromString` isn't efficient enough?

Reply:
UUID.fromString is too strict, yep.